modules/mod-deep-learning/ModelCardPanel.cpp

```diff
@@ -0,0 +1,518 @@
+      S.AddVariableText(Verbatim("%s")
+                            .Format(mCard->effect_type()));
+
+      S.AddVariableText(XO("Sample Rate: "))
+          ->SetFont(wxFont(wxFontInfo().Bold()));
+      S.AddVariableText(Verbatim("%d")
+                            .Format(mCard->sample_rate()));
+
+      std::string tagString;
+      for (auto &tag : mCard->tags())
+      {
+         if (!tagString.empty())
+            tagString = tagString +  ", " + tag;
+         else
+            tagString = tag;
+      }
+      
+      S.AddVariableText(XO("Tags: "))
+          ->SetFont(wxFont(wxFontInfo().Bold()));
+      S.AddVariableText(Verbatim("%s")
+                            .Format(tagString));
+   }
+   S.EndMultiColumn();
+}
+
+void ModelCardPanel::PopulateInstallCtrls(ShuttleGui &S)
+{
+   DeepModelManager &manager = DeepModelManager::Get();
+
+   S.StartVerticalLay(wxCENTER, true);
+   {
+      // add a progress gauge for downloads, but hide it
+      mInstallProgressGauge = safenew wxGauge(S.GetParent(), wxID_ANY, 100); // TODO:  sizing
+      mInstallProgressGauge->SetSize(wxSize(80, 20));
+      S.AddWindow(mInstallProgressGauge);
+
+      bool installed = manager.IsInstalled(mCard);
+      TranslatableString status = installed ? XO("installed") : XO("uninstalled");
+      mInstallStatusText = S.AddVariableText(status, true);
+
+      InstallStatus iStatus = installed ? InstallStatus::Installed : InstallStatus::Uninstalled;
+      wxColour statusColor = mInstallStatusColors[iStatus];
+      mInstallStatusText->SetForegroundColour(statusColor);
+
+      // TODO: do translatable strings here from the begginign
+      TranslatableString cmd = installed ? XO("Uninstall") : XO("Install");
+      mInstallButton = S.AddButton(cmd);
+
+      SetInstallStatus(installed ? InstallStatus::Installed : InstallStatus::Uninstalled);
+
+      mSelectButton = S.AddButton(XC("Select", "model"));
+      mSelectButton->Connect(wxEVT_COMMAND_BUTTON_CLICKED, 
+                        wxCommandEventHandler(ModelCardPanel::OnSelect), NULL, this);
+   }
+   S.EndVerticalLay();
+}
+
+void ModelCardPanel::PopulateMoreInfo(ShuttleGui &S)
+{
+   S.StartHorizontalLay(wxCENTER, true);
+   {
+      mMoreInfoButton = S.AddButton(XC("More Info", "model"));
+      mMoreInfoButton->Connect(wxEVT_COMMAND_BUTTON_CLICKED, 
+                           wxCommandEventHandler(ModelCardPanel::OnMoreInfo), NULL, this);
+   }
+}
+
+void ModelCardPanel::FetchModelSize()
+{
+   DeepModelManager &manager = DeepModelManager::Get();
+
+   ModelSizeCallback onGetModelSize = [this](size_t size)
+   {
+      this->CallAfter(
+         [this, size]()
+         {
+            float sizeMB = static_cast<float>(size) / static_cast<float>(1024 * 1024);
+            mModelSize->SetLabel(Verbatim("[%.1f MB]").Format(sizeMB).Translation());
+         }
+      );
+   }; 
+
+   manager.FetchModelSize(mCard, onGetModelSize);
+}
+
+void ModelCardPanel::SetInstallStatus(InstallStatus status)
+{
```

Comment:
Rather than all this retitling and rebinding of the buttons, you might find this solution more elegant:

Use `ShuttleGui` members `StartSimpleBook` and `EndSimplebook`.  In those brackets, use `StartNotebookPage` and `EndNotebookPage` one for each status.  And inside those brackets, add buttons and texts that don't need to change.  Then just use the `SetSelection` of the notebook control.

Reply:
Maybe the pages could even have different layouts, not every one of them having one button and one text.

Reply:
You would not need to show and hide the gauge, but just put it in one page.  Changing the notebook page is then enough to hide it.